arch/arm/core/aarch32/cortex_m/Kconfig

```diff
@@ -243,6 +243,14 @@ config ZERO_LATENCY_IRQS
 	  higher priority than the rest of the kernel they cannot use any
 	  kernel functionality.
 
+config DYNAMIC_DIRECT_INTERRUPTS
+	bool "Enable support for dynamic direct interrupts"
+	select DYNAMIC_INTERRUPTS
```

Comment:
@ulfalizer would be the authority but shouldn't we do "depends on DYNAMIC_INTERRUPTS" or "imply DYNAMIC_INTERRUPTS" instead?

"select" is kind of a bludgeon since it ignored dependencies

Reply:
If there's a chance that `DYNAMIC_INTERRUPTS` will have dependencies, I'd go with `depends on` at least. Currently it doesn't have any dependencies, so that makes it slightly less bad.

For `imply`, if you do `imply DYNAMIC_INTERRUPTS`, it will still be possible to disable `DYNAMIC_INTERRUPTS` when `DYNAMIC_DIRECT_INTERRUPTS` is enabled. That would be bad if it's required in that case.

Reply:
I suggest we change this to "depends on" then.

Reply:
If we change to depends-on, then one would need to set both Kconfig symbols, to have DYNAMIC_DIRECT_INTERRUPTS enabled, correct, @ulfalizer ?

If so, I am not sure I like it, but I am open to ideas :)